ddtrace/tracer/writer_test.go

```diff
@@ -0,0 +1,296 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package tracer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var _ traceWriter = &agentTraceWriter{}
+var _ traceWriter = &logTraceWriter{}
```

Comment:
```suggestion
func TestImplementsTraceWriter(_ *testing.T) {
    var _ traceWriter = (*agentTraceWriter)(nil)
    var _ traceWriter = (*logTraceWriter)(nil)
}
```
Just an idea.

Reply:
Nice Idea. I think I improved on it as well.